frontend-beeyond/src/app/modules/template/pages/template-detail/template-detail.component.ts

```diff
@@ -0,0 +1,30 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Template } from '../../../../shared/models/template.model';
+import { ApiService } from '../../../../core/services/api.service';
+
+@Component({
+  selector: 'app-template-detail',
+  templateUrl: './template-detail.component.html',
+  styleUrls: ['./template-detail.component.scss']
+})
+export class TemplateDetailComponent implements OnInit {
+  @Input() routePath = '';
+
+  template: Template;
+  id: number;
+  monacoEditorOptions = { language: 'yaml', scrollBeyondLastLine: false, readOnly: true };
+
+  constructor(
+    private router: Router,
+    private route: ActivatedRoute,
+    private apiService: ApiService
+  ) {}
+
+  ngOnInit(): void {
+    this.id = Number(this.route.snapshot.paramMap.get('id'));
```

Comment:
Turn this into and Resolver. For more information read the [documentation](https://angular.io/api/router/Resolve)